corehq/apps/cloudcare/static/cloudcare/js/form_entry/form_ui.js

```diff
@@ -806,6 +809,32 @@ hqDefine("cloudcare/js/form_entry/form_ui", function () {
         return itemsPerRow !== null ? Math.round(constants.GRID_COLUMNS / itemsPerRow) : constants.GRID_COLUMNS;
     };
 
+     function AddGroup (json, parent) {
```

Comment:
I think the approach of having a separate entry makes sense, but I don't love the duplication. Suggestions to reduce that:
* Move the logic in `newRepeat` into a static function that's part of `AddGroup` (making the repeat itself a param), and then call it both here and from `Repeat`.
* Use the knockout `template` binding to DRY up the HTML.

Reply:
Also mulling over the naming: maybe `AddFirstGroup`? It took a moment to figure out that this entry is only for user-generated repeats that are currently entry. A comment might also take care of this.

Reply:
@orangejenny Do you think it is worth pulling `newRepeat` out when we already planning to get rid of `Repeat` entirely? After the formplayer update is deployed `Repeat` will never get called.

Reply:
oh, if `Repeat` is going away, no, don't bother. What will that mean for the HTML?

Reply:
I assume at that point I would remove the html template for `Repeat`. Or do you mean something else?